docs/gen_api_ref.py

```diff
@@ -0,0 +1,40 @@
+"""Generate the API reference pages and mkdocs nav."""
+from pathlib import Path
+
+import mkdocs_gen_files
+
+# flax.linen.Module gives mkdocstrings problems, so we filter out anything
+# that explicitly subclasses it, as well as private methods (which is default)
+FILTER_OUT_MODULE = """    handler: python
+    selection:
+        filters:
+        - "!^_[^_]"
+        - "!Module"
```

Comment:
Will this match anything with "Module" in the name, as opposed to just something specifically called Module? I guess this isn't an issue atm but wondering if it should be more specific in case there are other uses of the term "Module" later for other things.

Reply:
From my limited familiarity with regex, I think "!Module" matches all strings that are not exactly the string "Module". (Let me know if that's not right)

Reply:
I must admit I also am not an expert on regexes. I always feel like I should feel more comfortable with them but I never do 😆

But I'm not sure about this. The exclamation mark doesn't really seem to be a standard regex thing from what I can tell (i.e. in python regexes ! just matches a ! character literal) but in this context it can clearly be used that way. But I think the interpretation would be that the regex "Module" would match anything containing "Module" anywhere, and thus "!Module" would match anything not containing Module anywhere. I think probably what you need is "!^Module$",  where the ^ and $ ensure that the Module match is the entire string rather than just anywhere in the middle.

Reply:
A quick test locally seems to confirm, so changed! Good catch